### PR TITLE
chore: more span pointer tests

### PR DESCRIPTION
**Pull request by apiarian-datadog**
## Motivation

Now that span pointers for s3 and dynamodb are going to be released and we're starting work on the javascript tracer, let's add more tests.

## Changes

Adding system tests for s3: copy object and complete multipart upload; and dynamod: put item, update item, delete item, batch write item, and transact write items.

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

**Comment by apiarian-datadog** [comment]
test failures seem to be unrelated to the changes here.

File: tests/serverless/span_pointers/utils.py [hunk 1]
```diff
@@ -36,9 +36,17 @@ def make_single_span_link_validator(
     """
 
     def validator(span):
+        print(span)
```
Comment (cbeauchesne) [inline tests/serverless/span_pointers/utils.py:39]:
```suggestion
        logger.info(span)
```

**Review by cbeauchesne — CHANGES_REQUESTED** [review]

File: tests/serverless/span_pointers/utils.py (hunk 1 above)
Comment (apiarian-datadog) [reply tests/serverless/span_pointers/utils.py]: it's gone!

**Review by nhulston — APPROVED** [review]
LGTM!

**Review by cbeauchesne — APPROVED** [review]

**Comment by apiarian-datadog** [comment]
remaining test failures are not relevant.

**Pull request merged**